build_tools/azure/posix-docker.yml

```diff
@@ -116,3 +118,12 @@ jobs:
         JUNIT_FILE: $(TEST_DIR)/$(JUNITXML)
       condition: and(succeededOrFailed(), eq(variables['CREATE_ISSUE_ON_TRACKER'], 'true'),
                      eq(variables['Build.Reason'], 'Schedule'))
+    - bash: echo "##vso[task.prependpath]$CONDA/bin"
+      displayName: Add conda to PATH
+    - bash: |
+        conda create -n $VIRTUALENV python=3.10 coverage -y
```

Comment:
I think it would be slightly simpler to move the `coverage combine --append` from `upload_codecov.sh` to `test_script.sh` this way you would not need to create a conda env here only to run this single line.

I don't expect this to be too much of a problem, but also it would make sure that `coverage` versions are consistent between the `pytest` run and `coverage combine`

Reply:
Yea that makes sense. I needed to make "combine coverage" it's own step because `test_pytest_soft_dependency.sh` also generates coverage after `test_script.sh`. The new `combine_coverage_reports.sh` runs after `test_pytest_soft_dependency.sh`.